### PR TITLE
[ Milestone 1 ] Feature setup

### PR DESCRIPTION
**Pull request by CakeInTech**
- Created a class Person with the following:
     -  Instance vars: @id, @name, and @age.
     -  Constructor with name, age, and parent_permission as parameter. name and 
         parent_permission are optional and have default values of "Unknown" and true.
     -  Getters for @id, @name, and @age.
     -  Setters for @name and @age.
     -  Private method is_of_age? that returns true if @age is greater or equal to 18 and false 
         otherwise.
    - Public method can_use_services? that returns true if person is of age or if they have permission from parents.
- Created a class Student with the following:
   - Inherits from Person.
   - Constructor extends parent's constructor by adding @classroom and a parameter for it.
   - Method play_hooky that returns "¯\(ツ)/¯".
- Create class Teacher with the following:
   - Inherits from Person.
   - Constructor extends parent's constructor by adding @specialization and a parameter for it.
   - Override can_use_services? so it always returns true.
   - Each class should be saved in a separate file.

### CODE REVIEW

**Comment by TedLivist** [comment]
## STATUS: INVALID CODE REVIEW REQUEST 🛑 🛑

Hi @MohamedCK,

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

### Kindly remove the Microverse logo on your README

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Comment by CakeInTech** [comment]
Hi @TedLivist, 

Thank you for the review but may I ask whey we are removing the microverse logo when it's already there on the template.

File: person.rb [hunk 1]
```diff
@@ -0,0 +1,21 @@
+class Person
+  attr_accessor :name, :age
+  attr_reader :id
+
+  def initialize(age, name = 'Unknown', parent_permission: true)
+    @id = generate_id
```
Comment (TedLivist) [inline person.rb:6]: - [x] Kindly define `generate_id` to generate the ID, or you can simply generate it quickly.

  ```
  @id = rand(1...1000)
  ```

File: person.rb [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    @name = name
+    @age = age
+    @parent_permission = parent_permission
+  end
+
+  def can_use_services?
+    is_of_age? || @parent_permission
```
Comment (TedLivist) [inline person.rb:13]: - [x] What you have is `of_age?`, not `is_of_age?`. Kindly fix that.

File: student.rb [hunk 3]
```diff
@@ -0,0 +1,10 @@
+class Student < Person
```
Comment (TedLivist) [inline student.rb:3]: - [x] For this inheritance to work you need to import the Person class into this file:

  ```
  require_relative 'person.rb'
  ```

File: student.rb [hunk 4]
```diff
@@ -0,0 +1,10 @@
...
+  def initialize(age, classroom, name = 'Unknown', parent_permission: true)
+    super(age, name, parent_permission)
```
Comment (TedLivist) [inline student.rb]: - [x] You need to define the `parent_permission` correctly in the `#super` method.

  ```
  super(age, name, parent_permission: parent_permission)
  ```

File: teacher.rb [hunk 5]
```diff
@@ -0,0 +1,10 @@
+class Teacher < Person
+  def initialize(age, specialization, name = 'Unknown', parent_permission: true)
+    super(age, name, parent_permission)
```
Comment (TedLivist) [inline teacher.rb]: - [x] You need to define the `parent_permission` correctly in the `#super` method.

  ```
  super(age, name, parent_permission: parent_permission)
  ```

File: teacher.rb [hunk 6]
```diff
@@ -0,0 +1,10 @@
+class Teacher < Person
```
Comment (TedLivist) [inline teacher.rb:3]: - [x] For this inheritance to work you need to import the Person class into this file:

  ```
  require_relative 'person.rb'
  ```

**Review by TedLivist — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ ♻️

Hi @MohamedCK,

Good job so far!
✔️ Good documentation
✔️ Implementation of classes in different files based on convention
However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: person.rb (hunk 2 above)
Comment (CakeInTech) [reply person.rb:13]: @TedLivist  I had the `Is_of_age` but I'm getting an error from the linters saying I have to use `of_age`  instead so how do I go on about this. 😄

**Comment by TedLivist** [comment]
Hi @MohamedCK the reason for removing the logo is that you are required to customize the template. One of the possible reasons the logo is present in the template is probably because the template is tailored as a completely professional template. It is only natural (and professional) that you use your own logo or remove the irrelevant logo completely.

**Review by Kossi-stack — APPROVED** [review]
## Status: Approved ✔️

**Hi @MohamedCK,**

You did a really great job! 💯 💯💯💯
Congratulations! 🎉 🎉 🎉 🎉 

128382030-38e7de91-f6f2-45cf-b54e-8068e0446e4c

You may close this Pull Request and merge the changes
Keep being awesome dev's and good luck with your next milestone!

## To Highlight

✅ Descriptive pull request message
✅ linter error checks validated
✅ Documented your work professionally
✅ README file is descriptive.
✅ Commit messages are meaningful.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding! 👏 👏 👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.**

As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using this [form.](https://airtable.com/shrQAqnBwek5a0O0s)

**Pull request merged**